### PR TITLE
R2 backports 2

### LINKED ISSUE #1422: Backport "Correct MOVD instruction and SSE registers definition"
radareorg/radare2@676cb56

### LINKED ISSUE #1424: Backport "System LIBRARY_PATH must be honored (not replaced) on HaikuOS"
radareorg/radare2@8abc24c

### LINKED ISSUE #1425: Backport "Fix coverity issue in Mach-O format"
radareorg/radare2@f47804f : Mach0 leak, to check

### LINKED ISSUE #1426: Backport "Fix large loading times when parsing encrypted/fuzzed macho"
radareorg/radare2@f0cafe9 : Mach0 issue

### LINKED ISSUE #1427: Backport "Dont trust the unaligned rich PEs ##bin"
radareorg/radare2@4a2e846 : PE issue, to check

### LINKED ISSUE #1428: Backport "Fix emulation for ARM's ldrd"
radareorg/radare2@c6f09b9 : analysis_arm_cs issue

### LINKED ISSUE #1430: Backport "Wrong assembly generated for: 'add x0, x0, 1, lsl # 12'"
radareorg/radare2@2b5bf5a

### PR DESCRIPTION
**Pull request by ret2libc**
**REBASE ME***

Close https://github.com/rizinorg/rizin/issues/1430 .
Close https://github.com/rizinorg/rizin/issues/1428
Close https://github.com/rizinorg/rizin/issues/1427
Close https://github.com/rizinorg/rizin/issues/1426
Close https://github.com/rizinorg/rizin/issues/1425
Close https://github.com/rizinorg/rizin/issues/1424
Close https://github.com/rizinorg/rizin/issues/1422

### CODE REVIEW

File: librz/socket/run.c [hunk 1]
```diff
@@ -1016,7 +1016,11 @@ RZ_API int rz_run_config_env(RzRunProfile *p) {
 #if __WINDOWS__
 		eprintf("rz-run: libpath unsupported for this platform\n");
 #elif __HAIKU__
-		rz_sys_setenv("LIBRARY_PATH", p->_libpath);
+		char *orig = r_sys_getenv("LIBRARY_PATH");
```
Comment (XVilka) [inline librz/socket/run.c]: `rz_`

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/socket/run.c (hunk 1 above)
Comment (ret2libc) [reply librz/socket/run.c]: oh, nice catch :D

**Comment by XVilka** [comment]
By the way, next time please use `import-patches` to trigger more CI jobs.

**Review by XVilka — APPROVED** [review]
Please fix the clang-format problem. Apart from that looks ok.

cc @devnexen regarding the HaikuOS patch - does it looks OK?

**Comment by devnexen** [comment]
yes it does.

**Pull request merged**